### PR TITLE
[BugFix] add scan statistics to auditlog in pipeline-mode

### LINKED ISSUE #5810: Lose scanRows of auditlog in pipeline mode
Lose several statistics of auditlog in pipeline mode, including `scanRows` and `scanBytes`.

### Steps to reproduce the behavior (Required)

In pipeline mode, query statistics in auditlog are empty, both `ScanRows` and `ScanBytes` are `0`.

```
2022-05-07 10:55:37,727 [query] |Client=127.0.0.1:42034|User=root|ResourceGroup=default_wg|Db=default_cluster:hehe|State=EOF|ErrorCode=|Time=22|ScanBytes=0|ScanRows=0|ReturnRows=1|StmtId=213|QueryId=2c41ae78-cdb1-11ec-af86-66660a909341|IsQuery=true|feIp=172.26.92.195|Stmt=select * from tj where id = 2|Digest=
```

The direct cause is `OlapScanNode::collect_query_statistics` would collect query statistic after execution, but in pipeline-mode there's no such behaviour.

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - main `61beccb99`

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5810

## Problem Summary(Required) ：

Add `scanRows` and `scanBytes` to auditlog in pipeline-mode.
- In non-pipe mode, the query statistic is collected bottom-up from scan node
- In pipe mode, the query statistic is recorded at `QueryContext` directly

### CODE REVIEW

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Review by stdpain — DISMISSED** [review]

File: be/src/exec/pipeline/pipeline_driver_queue.cpp [hunk 1]
```diff
@@ -110,6 +111,10 @@ size_t QuerySharedDriverQueue::size() {
 
 void QuerySharedDriverQueue::update_statistics(const DriverRawPtr driver) {
     _queues[driver->get_driver_queue_level()].update_accu_time(driver);
+    if (SourceOperator* source_operator = driver->source_operator()) {
```
Comment (kangkaisen) [inline be/src/exec/pipeline/pipeline_driver_queue.cpp:114]: exchange also is source，should check is scan?

Comment (ZiheLiu) [inline be/src/exec/pipeline/pipeline_driver_queue.cpp:114]: `QuerySharedDriverQueue::update_statistics` is used to update statistics for the driver queue. Maybe this logic is better to move to `PipelineDriver::_update_statistics`? In this way, we needn't write duplicated codes twice in `QuerySharedDriverQueue::update_statistics` and `DriverQueueWithWorkGroup::update_statistics`.

Comment (mofeiatwork) [reply be/src/exec/pipeline/pipeline_driver_queue.cpp:114]: The `scan_rows` and `scan_bytes` are maintained at `SourceOperator`, and currently only `OlapChunkSource` provide these two statistic, so the `Exchange` has no affect with it

Comment (mofeiatwork) [reply be/src/exec/pipeline/pipeline_driver_queue.cpp:114]: Good advice, I could move it the `PipelineDriver::process` and `PipelineDriver::_update_statistics`.

**Review by kangkaisen — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**